### PR TITLE
PYIC-3862: Upgrade the log into a correlation failure, unless we see any other legitimate instances where a VC is missing name/dob.

### PR DESCRIPTION
**Pull request by TalatAydinGDS**
### What changed

Upgrade the log into a correlation failure, unless we see any other legitimate instances where a VC is missing name/dob

### Why did it change

In accordance with GPG45 rules, except for exceptios, the 'name' and 'birthdate' should not empty or null. We throws exception to catch them.

### Issue tracking

- [PYIC-3862](https://govukverify.atlassian.net/browse/PYIC-3862)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3862]: https://govukverify.atlassian.net/browse/PYIC-3862?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java [hunk 1]
```diff
@@ -510,6 +511,9 @@ private List<IdentityClaim> getIdentityClaimsForBirthDateCorrelation(
                     continue;
                 }
                 addLogMessage(item, "Birthdate property is missing from VC");
+                throw new HttpResponseExceptionWithErrorBody(
```
Comment (rpayal) [inline libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java:514]: Is method isBirthDateEmpty() be private ?

Comment (MikeCollingwood) [reply libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java:514]: I think CredentialParseException could make sense here

**Review by rpayal — APPROVED** [review]
LGTM

File: libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java [hunk 2]
```diff
@@ -510,13 +511,16 @@ private List<IdentityClaim> getIdentityClaimsForBirthDateCorrelation(
...
+                        HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        ErrorResponse.FAILED_BIRTHDATE_CORRELATION);
```
Comment (Joe-Edwards-GDS) [inline libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java:516]: I hadn't really considered, but worth noting that this means we treat missing fields differently to other correlation failures - in practice meaning that a missing birthdate will result in a technical error, but an incorrect birthdate will result in PYI_NO_MATCH.

That seems ok to me though - if a CRI starts omitting one of these fields, it's a bug that we want to notice!

**Pull request merged**